packages/lesswrong/unitTests/autoRateLimit.tests.ts

```diff
@@ -67,13 +68,14 @@ function assignTotalDocumentKarma(votes: RecentVoteInfo[]) {
   }));
 }
 
-export function calculateFeatures(userId: string, userKarmaInfo: UserKarmaInfo, allVotes: RecentVoteInfo[]): RateLimitFeatures {
-  const recentKarmaInfo = calculateRecentKarmaInfo(userId, allVotes)
-  const features = {
-    ...recentKarmaInfo, 
-    downvoteRatio: getDownvoteRatio(userKarmaInfo)
-  }
-  return features
+export async function calculateFeatures(userId: string, userKarmaInfo: UserKarmaInfo, allVotes: RecentVoteInfo[]): Promise<RateLimitFeatures> {
```

Comment:
You are not going to be able to talk to the database inside of unit tests, I think.  (I could be wrong?  But I don't think we have that kind of setup right now.)